hoomd/mpcd/DryingDropletStreaming.h

```diff
@@ -0,0 +1,165 @@
+/*!
+ * This method implements the base version of ballistic propagation of MPCD
+ * particles in Spherical geometry.
+ *
+ * \tparam Geometry The confining geometry (SphereGeometry).
+ *
+ * The integration scheme is essentially Verlet with specular reflections. The particle is streamed forward over
+ * the time interval. If it moves outside the Geometry, it is placed back on the boundary and its velocity is
+ * updated according to the boundary conditions. Streaming then continues until the timestep is completed.
+ *
+ * To facilitate this, every Geometry must supply three methods:
+ *  1. detectCollision(): Determines when and where a collision occurs. If one does, this method moves the
+ *                        particle back, reflects its velocity, and gives the time still remaining to integrate.
+ *  2. isOutside(): Determines whether a particles lies outside the Geometry.
+ *  3. validateBox(): Checks whether the global simulation box is consistent with the streaming geometry.
+ *
+ */
+template<class Geometry>
+class PYBIND11_EXPORT DryingDropletStreaming : public mpcd::ConfinedStreamingMethod
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata MPCD system data
+         * \param cur_timestep Current system timestep
+         * \param period Number of timesteps between collisions
+         * \param phase Phase shift for periodic updates
+         * \param geom Streaming geometry
+         * \param R is the radius of sphere intially at very first timestep
+         */
+        DryingDropletStreaming(std::shared_ptr<mpcd::SystemData> sysdata,
+                                unsigned int cur_timestep,
+                                unsigned int period,
+                                int phase,
+                                std::shared_ptr<const Geometry> geom, Scalar R)
+        : mpcd::ConfinedStreamingMethod(sysdata, cur_timestep, period, phase),
+          m_geom(geom), m_validate_geom(true)
+          {}
+
+        //! Implementation of the streaming rule
+        virtual void stream(unsigned int timestep);
+
+        //! Get the streaming geometry
+        std::shared_ptr<const Geometry> getGeometry() const
+            {
+            return m_geom;
+            }
+
+        //! Set the streaming geometry
+        void setGeometry(std::shared_ptr<const Geometry> geom)
+            {
+            m_validate_geom = true;
+            m_geom = geom;
+            }
+
+    protected:
+        std::shared_ptr<const Geometry> m_geom; //!< Streaming geometry
+        bool m_validate_geom;   //!< If true, run a validation check on the geometry
+
+        //! Validate the system with the streaming geometry
+        void validate();
+
+        //! Check that particles lie inside the geometry
+        virtual bool validateParticles();
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+template<class Geometry>
+void DryingDropletStreaming<Geometry>::stream(unsigned int timestep)
+    {
+    
```

Comment:
This will look like:
```suggestion
    // update the geometry, only validating if this is the first time we set it
    const double start_R = (*m_R)(timestep);
    const double end_R = (*m_R)(timestep + m_period);
    const double V = (end_R - start_R)/(m_period * m_mpcd_dt);
    if (V > 0)
        {
        throw std::runtime_error("Droplet radius must decrease.");
        }
    m_validate_geom = m_geom;
    m_geom = std::make_shared<const mpcd::SphericalGeometry>(new_R, V);

    // stream using the parent method
    ConfinedStreamingMethod<mpcd::SphericalGeometry>::stream(timestep);

    // evaporate marked particles
    // ...
```